config/initializers/aws_cloudwatch.rb

```diff
@@ -0,0 +1,54 @@
+# gem install aws_sdk --version=3.1.5
+require 'aws-sdk-cloudwatchlogs'
+
+Rails.logger.debug "Running initializer in #{__FILE__}"
+
+def find_log_stream_name
```

Comment:
I'm pondering wrapping this whole thing in a timeout. @eanders @andyzito what do you two think?

Reply:
Sure. Is there a specific rails pattern you had in mind, or do you just mean adding a condition to bail out if too much time has elapsed?

Reply:
I was just thinking of `Timeout::timeout(15) do ... end` in the method

Reply:
@andyzito, this feels like it would still be good to have the Timeout in there.